ECS Task Observer endpoint implementation

rmfitzpatrick opened the pull request:
**Description:**
These changes implement the new ECS task observer extension defined by https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6121.  They don't adopt it as an available component by the service though.

**Link to tracking Issue:**
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/5316

**Testing:**
Unit tests added for provided functionality

**Documentation:**
Minor doc corrections but existing documentation is adequate.

pmcollins commented on extension/observer/ecstaskobserver/config_test.go:
```diff
@@ -0,0 +1,73 @@
+
+	factory := NewFactory()
+	factories.Extensions[typeStr] = factory
+	cfg, err := configtest.LoadConfigAndValidate(path.Join(".", "testdata", "config.yaml"), factories)
```
nit: do we need the dot (first arg to Join)?

rmfitzpatrick replied:
This is used 190+ times in this project but I think the whole convention is unnecessary overall.  I'm not sure why  `path` is used instead of `path/filepath` since it uses the os separator instead of hardcoded `/`.

mx-psi replied:
@rmfitzpatrick can you open an issue for this?

rmfitzpatrick replied:
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/6480

pmcollins commented on extension/observer/ecstaskobserver/extension.go:
```diff
@@ -15,21 +15,102 @@
 	telemetry        component.TelemetrySettings
 }
 
+func (e *ecsTaskObserver) Shutdown(ctx context.Context) error {
```
Wondering if `Shutdown` should go towards the end of the file since it's done last?

rmfitzpatrick replied:
Since this is the `Extension`/`Component` method I put it first by convention.

pmcollins reviewed (approved):
LGTM

mx-psi reviewed (changes requested):
You need to add yourself as a CODEOWNER to contribute this new component

Commit:
ECS Task Observer endpoint implementation

These changes implement the new ECS task observer extension defined by https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6121.  They don't adopt it as an available component by the service though.
